### PR TITLE
DT-2295 add topic config to queue lib

### PR DESCRIPTION
**Pull request by mikehalmamoj**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -121,7 +121,13 @@ It is only possible to use the `@SpyBean` annotation for beans declared in a `@C
 
 However, as mentioned above you can override the automatically generated beans with your own bean, e.g. with bean name `<queueId>-sqs-client` or `<queueId>-sqs-dlq-client`. If you do this in a `@TestConfiguration` using the `@Bean` annotation then it is possible to declare a corresponding SpyBean. `HmppsQueueFactory` provides factory methods to assist in creating such beans.
 
-An example in the `test-app` class which defines beans to spy on in a `@TestConfiguration`. See [IntegrationTestBase](https://github.com/ministryofjustice/hmpps-spring-boot-sqs/blob/main/test-app/src/test/kotlin/uk/gov/justice/digital/hmpps/hmppstemplatepackagename/integration/IntegrationTestBase.kt).
+This is complicated so first check the usage of your Spy Beans. Are they actually being used to mock or verify or just using them to purge queues / count messages on a queue? If there is no mocking or verifying then you can get the real bean from the `HmppsQueueService` by doing:
```
Comment (mikehalmamoj) [inline README.md:124]: unrelated readme change

File: build.gradle.kts [hunk 2]
```diff
@@ -10,7 +10,7 @@ plugins {
 
 allprojects {
   group = "uk.gov.justice.service.hmpps"
-  version = "0.6.0"
+  version = "0.7.0-beta"
```
Comment (mikehalmamoj) [inline build.gradle.kts:13]: This is still a work in progress so prevent accidental publish

File: lib/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsQueueFactory.kt [hunk 3]
```diff
@@ -76,6 +76,7 @@ class HmppsQueueFactory(
         "localstack" ->
           amazonSqsFactory.localStackSqsClient(queueConfig.queueName, localstackUrl, region, queueConfig.asyncQueueClient)
             .also { sqsClient -> createLocalStackQueue(sqsClient, sqsDlqClient, queueConfig.queueName, queueConfig.dlqName) }
+        // TODO for localstack we should subscribe queues to topics if configured
```
Comment (mikehalmamoj) [inline lib/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsQueueFactory.kt:79]: A few TODOs - coming in another PR

File: lib/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsQueueJmsListenerContainerFactory.kt [hunk 4]
```diff
@@ -18,6 +18,6 @@ class HmppsQueueJmsListenerContainerFactory(private val factories: List<HmppsQue
       .firstOrNull { it.destination == (endpoint as MethodJmsListenerEndpoint).destination }
       ?.factory
       ?.createListenerContainer(endpoint)
-      ?: throw JmsListenerContainerFactoryMissingException("Unable to find jms listener container factory for endpoint ${endpoint.id}")
+      ?: throw JmsListenerContainerFactoryMissingException("Unable to find jms listener container factory for endpoint ${(endpoint as MethodJmsListenerEndpoint).destination}")
```
Comment (mikehalmamoj) [inline lib/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsQueueJmsListenerContainerFactory.kt:21]: Unrelated fix to error message

File: lib/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsSqsProperties.kt [hunk 5]
```diff
@@ -3,25 +3,40 @@ package uk.gov.justice.hmpps.sqs
...
   val queues: Map<String, QueueConfig>,
+  val topics: Map<String, TopicConfig> = mapOf(),
 ) {
   data class QueueConfig(
     val queueName: String,
     val queueAccessKeyId: String = "",
     val queueSecretAccessKey: String = "",
     val asyncQueueClient: Boolean = false,
+    val subscribeTopicId: String = "",
+    val subscribeFilter: String = "",
```
Comment (mikehalmamoj) [inline lib/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsSqsProperties.kt:23]: Placeholders, not implemented yet

**Review by karenmillermoj — APPROVED** [review]
looks great
An additional test would be good to ensure multiple topics with the same name/arn can't be created
